File: src/design/data/DesignGenerationResult.ts
import type {DesignCandidateData} from './DesignCandidateData.js'
import type {DesignIntentData} from './DesignIntentData.js'
export interface DesignValidationData{readonly designDistancePassed:boolean;readonly browserValidated:boolean;readonly notes:readonly string[]}
export interface DesignGenerationResult{readonly version:1;readonly prompt:string;readonly intent:DesignIntentData;readonly candidates:readonly DesignCandidateData[];readonly validation:DesignValidationData}

File: src/main/java/org/tavall/webdesign/design/data/DesignGenerationResult.java
package org.tavall.webdesign.design.data;

import java.util.List;

public record DesignGenerationResult(
        int version,
        String prompt,
        DesignIntent intent,
        List<DesignCandidate> candidates,
        DesignValidation validation
) {
    public DesignGenerationResult {
        candidates = List.copyOf(candidates);
    }
}
